core/src/main/java/org/fao/geonet/kernel/AccessManager.java

```diff
@@ -183,6 +185,23 @@
         return hs;
     }
 
+    /**
+     *  Retrieves the user's groups ids
+     * @param session
+     * @param profile
+     * @return
+     * @throws SQLException
+     */
+    public static List<Integer> getGroups(UserSession session, Profile profile) throws SQLException {
+        ApplicationContext applicationContext = ApplicationContextHolder.get();
+        final UserGroupRepository userGroupRepository = applicationContext.getBean(UserGroupRepository.class);
```

Comment:
Maybe check to use `Autowire`? Similar in other methods that were already defined.

Reply:
Done for the whole file....